xapian-letor/ranker/ranker.cc

```diff
@@ -196,12 +196,15 @@ static std::pair<string, string>
 parse_query_string(const string & query_line, int line_number)
 {
     vector<string> token;
-    size_t j = 0;
-    while (j < query_line.size()) {
-	size_t i = query_line.find_first_not_of(' ', j);
-	if (i == string::npos) break;
-	j = query_line.find_first_of(' ', i);
-	token.push_back(query_line.substr(i, j - i));
+    size_t i = 0;
+    size_t j = query_line.find_first_of(' ', i);
+    token.push_back(query_line.substr(i, j - i));
```

Comment:
Actually, that'll work as written - `string::npos` is the maximum value of `string::size_type`, so if `j == string::npos` this will ask for a substring which is >= the length of the string and that will just return the rest of the string.  That's perhaps not entirely clear to the reader without some thought though.

Also, `i` is always zero here and only really used when looking for single quotes.